tests/integration/src/test/java/org/mule/test/core/context/notification/processors/ModuleComponentPathTestCase.java

```diff
@@ -0,0 +1,310 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.test.core.context.notification.processors;
+
+import static java.util.Arrays.asList;
+import static java.util.Optional.empty;
+import static java.util.Optional.of;
+import static org.hamcrest.core.Is.is;
+import static org.mule.runtime.api.component.ComponentIdentifier.buildFromStringRepresentation;
+import static org.mule.runtime.api.component.TypedComponentIdentifier.ComponentType.FLOW;
+import static org.mule.runtime.api.component.TypedComponentIdentifier.ComponentType.PROCESSOR;
+import static org.mule.runtime.api.component.TypedComponentIdentifier.builder;
+import static org.mule.runtime.api.dsl.DslResolvingContext.getDefault;
+import static org.mule.runtime.config.spring.dsl.model.ApplicationModel.FLOW_IDENTIFIER;
+import static org.mule.runtime.core.api.lifecycle.LifecycleUtils.initialiseIfNeeded;
+import org.junit.Assert;
+import org.junit.Test;
+import org.mule.functional.junit4.MuleArtifactFunctionalTestCase;
+import org.mule.runtime.api.component.TypedComponentIdentifier;
+import org.mule.runtime.api.dsl.DslResolvingContext;
+import org.mule.runtime.api.meta.model.ExtensionModel;
+import org.mule.runtime.core.DefaultMuleContext;
+import org.mule.runtime.core.api.MuleContext;
+import org.mule.runtime.core.api.config.ConfigurationBuilder;
+import org.mule.runtime.core.config.builders.AbstractConfigurationBuilder;
+import org.mule.runtime.core.context.notification.MessageProcessorNotification;
+import org.mule.runtime.dsl.api.component.config.DefaultComponentLocation;
+import org.mule.runtime.extension.internal.loader.XmlExtensionModelLoader;
+import org.mule.runtime.module.extension.internal.manager.DefaultExtensionManager;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Test suite to ensure XML connectors do properly generate the paths
+ *
+ * @since 4.0
+ */
+public class ModuleComponentPathTestCase extends MuleArtifactFunctionalTestCase {
```

Comment:
What should I put? the siblings of this test do not have that usage either (`org.mule.test.core.context.notification.processors.ComponentPathTestCase`, neither any subclass of `org.mule.test.core.context.notification.processors.AbstractMessageProcessorNotificationTestCase` )